dotCMS/src/integration-test/java/com/dotmarketing/servlets/BinaryExporterServletTest.java

```diff
@@ -0,0 +1,213 @@
+import com.dotcms.mock.request.MockHttpRequest;
+import com.dotcms.mock.request.MockServletPathRequest;
+import com.dotcms.mock.request.MockSessionRequest;
+import com.dotcms.mock.response.BaseResponse;
+import com.dotcms.mock.response.MockHttpCaptureResponse;
+import com.dotcms.mock.response.MockHttpStatusResponse;
+import com.dotcms.util.CollectionsUtils;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Permission;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.model.User;
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Collection;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class BinaryExporterServletTest {
+
+    // Temporary binary png file
+    private static class TmpBinaryFile implements Closeable {
+
+        private final Path pngFilePath;
+
+        public TmpBinaryFile(final boolean setContent) throws IOException {
+            pngFilePath = Files.createTempFile("tmp", ".png");
+            if (setContent) {
+                Files.write(pngFilePath, ShortyServletAndTitleImageTest.pngPixel);
+            }
+        }
+
+        public Path getPath() {
+            return pngFilePath;
+        }
+
+        public File getFile() {
+            return pngFilePath.toFile();
+        }
+
+        @Override
+        public void close() throws IOException {
+            Files.deleteIfExists(pngFilePath);
+        }
+    }
+
+    private static final String BY_ID = "by-identifier";
+    private static final String BY_INODE = "by-inode";
+
+    private static final String READ_PERMISSIONS = "has-read-permissions";
+    private static final String NO_PERMISSIONS = "no-permissions";
+
+    private static Host host;
+    private static Role role;
+
+    final private boolean byIdentifier;
+    final private boolean permissionsRequired;
+
+    public BinaryExporterServletTest(final String byIdType, final String permissionType) {
+        this.byIdentifier = byIdType.equals(BY_ID);
+        this.permissionsRequired = permissionType.equals(NO_PERMISSIONS);
+    }
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4885343)